### DISCUSSION TITLE
Calling pep702 deprecated directly

### DISCUSSION THREAD

**Discussion by jenshnielsen** [microsoft/pyright#7456] [Q&A] [answered]
This is mostly a question of the expected behaviour. 

I have been renaming a some functions as a part of a refactor but want to retain the original function names as deprecated aliases using PEP702 decorators. 

Imagine that I have the following function:

```python
def function_a() -> str:
    return "function_a_output"
```

I can now see two obvious ways of creating a deprecated alias

```python
from typing_extensions import deprecated

@deprecated("Use function_a")
def function_b() -> str:
     return function_a()
```

or 

```python
from typing_extensions import deprecated

function_b = deprecated("Use function_a")(function_a)
```

Both forms will work correctly at runtime calling the function with a deprecation warning and returning the expected result.

However, the second form (directly passing the function to the decorator) does not trigger a pyright warning at the site of function execution. My question is therefor. Is this expected? Is it mean to be supported to call the decorator directly? In some cases this would be preferable to me as I can directly copy a function with a different name avoiding to replicate the signature in an alias but perhaps I am overlooking something. 

Tested in VSCode with the latest pylance

**Comment by erictraut** [comment, marked as answer]
The current behavior is intended. PEP 702 indicates that `deprecated` is intended to be used as a decorator. If you use it in other ways (not as a decorator), it may work at runtime, but type checkers and language servers will not honor it.

**Reply by jenshnielsen** [reply]
Ok that's a shame but thanks for the answer.

**Reply by astafan8** [reply]
Would it be an idea to allow this kind of usage and pyright's treating it correctly via a pyright config entry?

**Reply by erictraut** [reply]
This isn't the sort of thing that deserves a config setting. The behavior should be implemented in accordance with the way the spec indicates it should work.

If you want to change the way that `deprecated` works, it would need to be done as part of the type specification update process. For details about this process, see [this page](https://github.com/python/typing-council#decisions).
